src/glmtools.jl

```diff
@@ -490,23 +490,63 @@ The loglikelihood of a fitted model is the sum of these values over all the obse
 """
 function loglik_obs end
 
-loglik_obs(::Bernoulli, y, μ, wt, ϕ) = wt*logpdf(Bernoulli(μ), y)
-loglik_obs(::Binomial, y, μ, wt, ϕ) = logpdf(Binomial(Int(wt), μ), _safe_int(y*wt))
-loglik_obs(::Gamma, y, μ, wt, ϕ) = wt*logpdf(Gamma(inv(ϕ), μ*ϕ), y)
+loglik_obs(::Bernoulli, y, μ, wt, ϕ) = wt * logpdf(Bernoulli(μ), y)
+loglik_obs(::Binomial, y, μ, wt, ϕ) = logpdf(Binomial(Int(wt), μ), _safe_int(y * wt))
+loglik_obs(::Gamma, y, μ, wt, ϕ) = wt * logpdf(Gamma(inv(ϕ), μ * ϕ), y)
 # In Distributions.jl, a Geometric distribution characterizes the number of failures before
 # the first success in a sequence of independent Bernoulli trials with success rate p.
 # The mean of Geometric distribution is (1 - p) / p.
 # Hence, p = 1 / (1 + μ).
 loglik_obs(::Geometric, y, μ, wt, ϕ) = wt * logpdf(Geometric(1 / (μ + 1)), y)
-loglik_obs(::InverseGaussian, y, μ, wt, ϕ) = wt*logpdf(InverseGaussian(μ, inv(ϕ)), y)
-loglik_obs(::Normal, y, μ, wt, ϕ) = wt*logpdf(Normal(μ, sqrt(ϕ)), y)
-loglik_obs(::Poisson, y, μ, wt, ϕ) = wt*logpdf(Poisson(μ), y)
+loglik_obs(::InverseGaussian, y, μ, wt, ϕ) = wt * logpdf(InverseGaussian(μ, inv(ϕ)), y)
+loglik_obs(::Normal, y, μ, wt, ϕ) = wt * logpdf(Normal(μ, sqrt(ϕ)), y)
+loglik_obs(::Poisson, y, μ, wt, ϕ) = wt * logpdf(Poisson(μ), y)
 # We use the following parameterization for the Negative Binomial distribution:
 #    (Γ(θ+y) / (Γ(θ) * y!)) * μ^y * θ^θ / (μ+θ)^{θ+y}
 # The parameterization of NegativeBinomial(r=θ, p) in Distributions.jl is
 #    Γ(θ+y) / (y! * Γ(θ)) * p^θ(1-p)^y
 # Hence, p = θ/(μ+θ)
 function loglik_obs(d::NegativeBinomial, y, μ, wt, ϕ)
-    return wt*logpdf(NegativeBinomial(d.r, d.r/(μ+d.r)),
-                     y)
+    return wt * logpdf(NegativeBinomial(d.r,
+                                        d.r / (μ + d.r)), y)
+end
+
+## Slight different interface for analytic and probability weights
+## ϕ is the deviance - not the deviance/n nor sum(wt)
+## sumwt is sum(wt)
+## n is the number of observations
+function loglik_apweights_obs(::Bernoulli, y, μ, wt, ϕ, sumwt, n)
```

Comment:
I just had a question: why can't this just be `loglik_obs` but dispatch on the type of weights? That is, why not just write:
```
loglik_obs(::Bernoulli, y,  μ, wt::AnalyticWeights, ϕ, sumwt, n)
```